Index: libchrome/crypto/scoped_openssl_types.h
===================================================================
--- libchrome.orig/crypto/scoped_openssl_types.h
+++ libchrome/crypto/scoped_openssl_types.h
@@ -45,20 +45,25 @@ struct OpenSSLFree {
 // Several typedefs are provided for crypto-specific primitives, for
 // short-hand and prevalence.
 using ScopedBIGNUM = ScopedOpenSSL<BIGNUM, BN_free>;
-using ScopedEC_Key = ScopedOpenSSL<EC_KEY, EC_KEY_free>;
+// Deprecated in OpenSSL 3.0
+//using ScopedEC_Key = ScopedOpenSSL<EC_KEY, EC_KEY_free>;
 using ScopedBIO = ScopedOpenSSL<BIO, BIO_free_all>;
-using ScopedDSA = ScopedOpenSSL<DSA, DSA_free>;
+// Deprecated in OpenSSL 3.0
+//using ScopedDSA = ScopedOpenSSL<DSA, DSA_free>;
 using ScopedECDSA_SIG = ScopedOpenSSL<ECDSA_SIG, ECDSA_SIG_free>;
 using ScopedEC_GROUP = ScopedOpenSSL<EC_GROUP, EC_GROUP_free>;
-using ScopedEC_KEY = ScopedOpenSSL<EC_KEY, EC_KEY_free>;
+// Deprecated in OpenSSL 3.0
+//using ScopedEC_KEY = ScopedOpenSSL<EC_KEY, EC_KEY_free>;
 using ScopedEC_POINT = ScopedOpenSSL<EC_POINT, EC_POINT_free>;
 using ScopedEVP_CIPHER_CTX = ScopedOpenSSL<EVP_CIPHER_CTX, EVP_CIPHER_CTX_free>;
 using ScopedEVP_MD_CTX = ScopedOpenSSL<EVP_MD_CTX, EVP_MD_CTX_free>;
 using ScopedEVP_PKEY = ScopedOpenSSL<EVP_PKEY, EVP_PKEY_free>;
 using ScopedEVP_PKEY_CTX = ScopedOpenSSL<EVP_PKEY_CTX, EVP_PKEY_CTX_free>;
-using ScopedHMAC_CTX = ScopedOpenSSL<HMAC_CTX, HMAC_CTX_free>;
+// Deprecated in OpenSSL 3.0
+//using ScopedHMAC_CTX = ScopedOpenSSL<HMAC_CTX, HMAC_CTX_free>;
 using ScopedNETSCAPE_SPKI = ScopedOpenSSL<NETSCAPE_SPKI, NETSCAPE_SPKI_free>;
-using ScopedRSA = ScopedOpenSSL<RSA, RSA_free>;
+// Deprecated in OpenSSL 3.0
+//using ScopedRSA = ScopedOpenSSL<RSA, RSA_free>;
 using ScopedX509 = ScopedOpenSSL<X509, X509_free>;
 
 // The bytes must have been allocated with OPENSSL_malloc.
